api/workload/v2/workload.go

```diff
@@ -0,0 +1,160 @@
+package workload
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/proto/api/workload"
+)
+
+const (
+	// DefaultAgentAddress is the default GRPC address to contact the spire agent at.
+	DefaultAgentAddress = "unix:///tmp/agent.sock"
+)
+
+// X509SVIDs is an X.509 SVID response from the SPIFFE Workload API.
+type X509SVIDs struct {
+	// SVIDs is a list of X.509 SVIDs.
+	SVIDs []*X509SVID
+}
+
+// Default returns the default SVID (the first in the list).
+func (x *X509SVIDs) Default() *X509SVID {
+	return x.SVIDs[0]
+}
+
+// SVID is an X.509 SPIFFE Verifiable Identity Document.
+//
+// See https://github.com/spiffe/spiffe/blob/master/standards/X509-SVID.md
+type X509SVID struct {
+	SPIFFEID     string
+	PrivateKey   crypto.Signer
+	Certificates []*x509.Certificate
+	TrustBundle  *x509.CertPool
+}
+
+// WorkloadIdentityWatcher is implemented by consumers who wish to be updated on SVID changes.
+type WorkloadIdentityWatcher interface {
+	// UpdateX509SVIDs indicates to the Watcher that the SVID has been updated
+	UpdateX509SVIDs(*X509SVIDs)
+
+	// OnError indicates an error occurred.
+	OnError(err error)
+
+	// OnConnection indicates a change in the connection state to the SPIFFE agent.
+	OnConnection(connected bool)
```

Comment:
In the past I've used a string to represent the remote peer here. `""` means disconnected. Having the peer info directly provided to the watcher makes more sense when the "peer" is a DNS string and you want to know what you are actually connected to. In a unix socket world it may be less useful but it's worth considering.